src/backend/distributed/commands/database.c

```diff
@@ -242,3 +262,458 @@ PreprocessAlterDatabaseSetStmt(Node *node, const char *queryString,
 
 	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
 }
+
+
+List *
+PreprocessCreateDatabaseStmt(Node *node, const char *queryString,
+							 ProcessUtilityContext processUtilityContext)
```

Comment:
I know this is the case but lastly we agreed on doing as much as we can in Postprocess?

**However, I'm also okay with having the checks in Preprocess for CREATE DATABASE** as it feels safer to ensure them before creating the database. This is because, we're not so sure about what does it mean to throw an error after creating a database in the same (implicit) transaction.

So let's keep it as is then.